### PR TITLE
log: dissallow nil loggers

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: abci/client/local_client.go [hunk 1]
```diff
@@ -26,15 +27,15 @@ var _ Client = (*localClient)(nil)
 // methods of the given app.
 //
 // Both Async and Sync methods ignore the given context.Context parameter.
-func NewLocalClient(mtx *sync.Mutex, app types.Application) Client {
+func NewLocalClient(logger log.Logger, mtx *sync.Mutex, app types.Application) Client {
 	if mtx == nil {
 		mtx = new(sync.Mutex)
```
Comment (creachadair) [inline abci/client/local_client.go:32]: Not related to your change, but yowzers. 🤨

File: libs/service/service.go [hunk 2]
```diff
@@ -104,10 +104,6 @@ type BaseService struct {
 
 // NewBaseService creates a new BaseService.
 func NewBaseService(logger log.Logger, name string, impl Implementation) *BaseService {
-	if logger == nil {
```
Comment (creachadair) [inline libs/service/service.go:107]: Since we're disallowing unset loggers, this could probably be a productive place for a panic. YMMV.

**Review by creachadair — APPROVED** [review]

File: libs/service/service.go (hunk 2 above)
Comment (tychoish) [reply libs/service/service.go:107]: I think it'll panic pretty quick (and te tests would catch this..) also there's no place in the code where we pass a nil logger, so I feel like it's ok to just let the panic happen for a nil pointer exception when we get there.

File: abci/client/local_client.go (hunk 1 above)
Comment (tychoish) [reply abci/client/local_client.go:32]: lol, yes.

**Pull request merged**